mycore-pi/pom.xml

```diff
@@ -120,11 +120,6 @@
       <groupId>org.mycore</groupId>
       <artifactId>mycore-jobqueue</artifactId>
     </dependency>
-    <dependency>
-      <groupId>org.mycore</groupId>
-      <artifactId>mycore-user2</artifactId>
-      <optional>true</optional>
```

Comment:
I think that has to be a optional runtime dependency. See [Maven Optional](https://maven.apache.org/guides/introduction/introduction-to-optional-and-excludes-dependencies.html).

Reply:
I don't see why that dependency would be needed anymore. With this PR, no direct references to classes from `mycore-user2` exist in `mycore-pi` anymore. Using `MCRUserInformationResolver`  effectively decouples `mycore-pi` from `mycore-user2`. I'm pretty sure that the Maven dependency plugin would rightfully flag `mycore-user2` as an unused dependency and fail the build, if I kept it as an dependency.